sdk/openai/inference/src/clients/openai_client.rs

```diff
@@ -0,0 +1,69 @@
+use std::sync::Arc;
+
+use azure_core::{Policy, Result, Url};
+
+use crate::{auth::OpenAIKeyCredential, OpenAIClientOptions};
+
+use super::{BaseOpenAIClientMethods, ChatCompletionsClient};
+
+/// Defines the methods provided by a [`OpenAIClient`] and can be used for mocking.
+pub trait OpenAIClientMethods {
+    fn chat_completions_client(&self) -> ChatCompletionsClient;
+}
+
+/// An OpenAI client.
+#[derive(Debug, Clone)]
+pub struct OpenAIClient {
+    base_url: Url,
+    pipeline: azure_core::Pipeline,
+    #[allow(dead_code)]
+    options: OpenAIClientOptions,
```

Comment:
I recommend not holding onto these if you can avoid it. Most should be involved in constructing the pipeline. Anything else you need out of it I would store *only* that on the client. Keep in mind clients must be stateless, however.